modules/ingest-download-service-impl/src/main/java/org/opencastproject/ingestdownloadservice/impl/endpoint/IngestDownloadServiceEndpoint.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ *
+ * The Apereo Foundation licenses this file to you under the Educational
+ * Community License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package org.opencastproject.ingestdownloadservice.impl.endpoint;
+
+import org.opencastproject.ingestdownloadservice.api.IngestDownloadService;
+import org.opencastproject.job.api.JaxbJob;
+import org.opencastproject.job.api.Job;
+import org.opencastproject.job.api.JobProducer;
+import org.opencastproject.mediapackage.MediaPackage;
+import org.opencastproject.mediapackage.MediaPackageParser;
+import org.opencastproject.rest.AbstractJobProducerEndpoint;
+import org.opencastproject.serviceregistry.api.ServiceRegistry;
+import org.opencastproject.serviceregistry.api.ServiceRegistryException;
+import org.opencastproject.util.doc.rest.RestParameter;
+import org.opencastproject.util.doc.rest.RestQuery;
+import org.opencastproject.util.doc.rest.RestResponse;
+import org.opencastproject.util.doc.rest.RestService;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+/**
+ * The REST endpoint for the {@link IngestDownloadService} service
+ */
+@Path("/")
+@RestService(name = "IngestDownloadServiceEndpoint",
+    title = "Ingest download REST endpoint",
+    abstractText = "The REST endpoint for the ingest download service.",
+    notes = {})
+public class IngestDownloadServiceEndpoint extends AbstractJobProducerEndpoint {
+  /** The logger */
+  private static final Logger logger = LoggerFactory.getLogger(IngestDownloadServiceEndpoint.class);
+
+  /** The service */
+  private IngestDownloadService service;
+  private ServiceRegistry serviceRegistry;
+
+  @POST
+  @Path("ingestdownload")
+  @Produces(MediaType.TEXT_XML)
+  @RestQuery(name = "ingestdownload",description = "Downloads mediapackage elements to workspace",
+          restParameters = { @RestParameter(description = "mediapackage as xml", isRequired = true, name = "mediapackage",
```

Comment:
I feel like these docs need work.  A few don't specify what you're separating by, and the booleans don't specify what they actually do.

Reply:
done. :)